devito/passes/clusters/factorization.py

```diff
@@ -64,8 +64,8 @@ def collect_const(expr):
 
     terms = []
     for k, v in inverse_mapper.items():
-        if len(v) == 1:
-            # We can actually evaluate everything to avoid, e.g., (-1)*a
+        if len(v) == 1 and not v[0].is_Add:
```

Comment:
note for reviewers: minor bug fix orthogonal to the main purpose of the PR

Reply:
Out of curiosity. What was the bug in this case?

Reply:
breaking a pre-factorized expression

```a(b+c)```

becoming

```ab + ac```